flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/jsonplan/JsonPlanGenerator.java

```diff
@@ -170,4 +174,58 @@ public static String generatePlan(
             throw new RuntimeException("Failed to generate plan", e);
         }
     }
+
+    public static String generateJsonStreamGraph(
+            ImmutableStreamGraph sg,
```

Comment:
I think using the `StreamGraph` is sufficient after we update it and ensure that this method is only called by the `AdaptiveGraphManager`

Reply:
Yes, it has been updated.